Sources/AuthFoundation/Token Management/Internal/KeychainTokenStorage.swift

```diff
@@ -46,10 +46,16 @@ final class KeychainTokenStorage: TokenStorage {
     
     var allIDs: [String] {
         do {
-            return try Keychain
+            let itemIDs = try Keychain
+                .Search(service: KeychainTokenStorage.serviceName)
+                .list()
+                .sorted(by: { $0.creationDate < $1.creationDate })
+                .map(\.account)
+            let metadataIDs = try Keychain
                 .Search(service: KeychainTokenStorage.metadataName)
                 .list()
                 .map(\.account)
+            return itemIDs.filter { metadataIDs.contains($0) }
```

Comment:
Does this prevent the `find` function from short circuiting for users already missing items/tokens?

Reply:
@baksha97 correct, this ensures that only tokens that have both their metadata and item keychain entries available to the device will be returned.  Therefore, the `find` and other functions will succeed.

This will still allow legitimate errors (such as an unsuccessful biometric prompt) to flow through.